java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -1467,6 +1467,7 @@ public ColumnView replaceChildrenWithViews(int[] indices,
           throw new IllegalArgumentException("Child row count doesn't match the old child");
         }
         newChildren.add(view);
+        child.close();
```

Comment:
This only covers one of the code paths where a new `ColumnView` was obtained.  What about the case where the view is added to `newChildren` just above this?  Also the view isn't always closed if an exception is thrown between when it is created and closed.

Reply:
The code path where we are adding `child` to `newChildren` doesn't need to be closed as the ownership of the `child` will be taken over by the new `ColumnView` created in this method. The call to `getChildColumnView()` doesn't create a new ColumnView, it wraps the existing one from how I read it because of this [line](https://github.com/rapidsai/cudf/blob/c2c67de0a487ef767f2b16e6b95132121e2eec04/java/src/main/native/src/ColumnViewJni.cpp#L1567) taking a reference and wrapping it. Even when the new ColumnView is created the CPP code doesn't create a new ColumnView for the child, it wraps existing child PTAL [here](https://github.com/rapidsai/cudf/blob/c2c67de0a487ef767f2b16e6b95132121e2eec04/cpp/src/column/column_view.cpp#L89). I may be reading the CPP code wrong, please let me know. 

>  Also the view isn't always closed if an exception is thrown between when it is created and closed.

Thanks, updated

Reply:
> The call to getChildColumnView() doesn't create a new ColumnView

This is incorrect.  It is heap-allocating a new `cudf::column_view`, that's what `std::make_unique<cudf::column_view>` does.  It's the heap allocation that needs to be released by closing the Java `ColumnView` instance.

> The code path where we are adding child to newChildren doesn't need to be closed as the ownership of the child will be taken over by the new ColumnView created in this method.

This is also incorrect.  The heap-allocated child view instances that are passed down to the JNI [are copied into the newly created view instance](https://github.com/rapidsai/cudf/blob/branch-0.20/java/src/main/native/src/ColumnViewJni.cpp#L1415) rather than being referenced directly, so they need to be closed separately.